bundles/org.openhab.binding.linkplay/src/main/java/org/openhab/binding/linkplay/internal/discovery/LinkPlayUpnpDiscoveryParticipant.java

```diff
@@ -0,0 +1,139 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A UPnP discovery participant for LinkPlay devices.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = UpnpDiscoveryParticipant.class)
+public class LinkPlayUpnpDiscoveryParticipant implements UpnpDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(LinkPlayUpnpDiscoveryParticipant.class);
+
+    private static final int TTL_SECONDS = 300;
+    private static final ServiceId SERVICE_ID_AV_TRANSPORT = new UDAServiceId("AVTransport");
+    private static final ServiceId SERVICE_ID_RENDERING_CONTROL = new UDAServiceId("RenderingControl");
+
+    private final HttpClient httpClient;
+
+    @Activate
+    public LinkPlayUpnpDiscoveryParticipant() {
+        try {
+            httpClient = new HttpClient(new SslContextFactory.Client(true));
+            httpClient.start();
+        } catch (Exception e) {
+            throw new IllegalStateException("Could not create HTTP client", e);
+        }
+    }
+
+    @Deactivate
+    public void deactivate() {
+        try {
+            httpClient.stop();
+        } catch (Exception e) {
+            logger.debug("Failed to stop HTTP client: {}", e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return LinkPlayBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(RemoteDevice device) {
+        if (!hasRequiredServices(device)) {
+            return null;
+        }
+        String udnIdentifier = device.getIdentity().getUdn().getIdentifierString();
+        if (udnIdentifier == null || udnIdentifier.isEmpty()) {
+            return null;
+        }
+        String udn = udnIdentifier.replace("uuid:", "").replace("-", "");
+        return new ThingUID(LinkPlayBindingConstants.THING_TYPE_PLAYER, udn);
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(RemoteDevice device) {
+        ThingUID thingUID = getThingUID(device);
+        if (thingUID == null) {
+            return null;
+        }
+
+        String host = device.getIdentity().getDescriptorURL().getHost();
+        if (host == null || host.isEmpty()) {
+            logger.trace("no host for device {}", device.getDetails().getFriendlyName());
+            return null;
+        }
+
+        Integer port = LinkPlayConnectionUtils.testConnection(httpClient, host);
+        if (port == null) {
+            return null;
+        }
+
+        String friendlyName = device.getDetails().getFriendlyName();
+        String deviceUDN = device.getIdentity().getUdn().getIdentifierString().replace("uuid:", "");
+
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(LinkPlayBindingConstants.CONFIG_UDN, deviceUDN);
+
+        String label = String.format("LinkPlay: %s", friendlyName);
```

Comment:
Not sure what friendlyname can be. Usually we prefer not to have the binding id as part of the label. as the thing is already boudn to the specific binding, there is no point / duplicate information.

Reply:
`friendylyName` is a UPnP term (so I'm pretty sure that's what it means here). It's a value provided by the device, and what's "friendly" about it is that it's formatted/presented in a "user-friendly way". It will usually be quite generic, but present the device and/or model.

Reply:
The truth is that there's no system to that. It's whatever the manufacturer think is a "suitable name" to present the device to consumers. It varies quite a lot. Usually, the manufacturer is there, model varies more in my experience, but I'm sure you can find examples of the opposite.

Reply:
Friendly name will be something like "Bedroom" or "Kitchen" ,  or whatever the user named it, so appending the binding id is very helpful when viewing among other things (IMHO)